uploader/pantheon.py

```diff
@@ -325,27 +354,58 @@ def processRegexMatches(files, globs, filetype):
     sys.exit('server ' + server + ' is not reachable')
 
 _info('Using server: ' + server)
-_info('Using ' + mode + ': ' + repository)
-print('--------------')
-
-moduleGlobs = readYamlGlob(config, 'modules')
-resourceGlobs = readYamlGlob(config, 'resources')
-non_resource_files = []
-logger.debug('moduleGlobs: %s', moduleGlobs)
-logger.debug('resourceGlobs: %s', resourceGlobs)
-logger.debug('args.directory: %s', args.directory)
-
-# List all files in the directory
-allFiles = []
-listdir_recursive(args.directory, allFiles)
-
-processRegexMatches(allFiles, resourceGlobs, 'resources')
-processRegexMatches(allFiles, moduleGlobs, 'modules')
-
-leftoverFiles = len(allFiles)
-if leftoverFiles > 0:
-    _warn(f'{leftoverFiles} additional files detected but not uploaded. Only files specified in '
-          + CONFIG_FILE
-          + ' are handled for upload.')
+
+if len(config.keys()) > 0 and 'repositories' in config:
+    for repo_list in config['repositories']:
+        repository = resolveOption(args.repository, '', repo_list['name'])
+        # Enforce a repository being set in the pantheon.yml
+        if repository == "" and mode == 'repository':
+            sys.exit('repository is not set')
+
+        mode = 'sandbox' if args.sandbox else 'repository'
+        # override repository if sandbox is chosen (sandbox name is the user name)
+        if args.sandbox:
+            repository = args.user
+
+        if 'attributes' in repo_list:
+            attributeFile = resolveOption(args.attrFile, '', repo_list['attributes'])
+        else:
+            attributeFile = resolveOption(args.attrFile, '', '')
+
+        if attributeFile and not os.path.isfile(attributeFile.strip()):
+            sys.exit('attributes: ' + attributeFile + ' does not exist.')
+
+        _info('Using ' + mode + ': ' + repository)
+        _info('Using attributes: ' + attributeFile)
+        print('--------------')
+
+        if attributeFile:
+            process_workspace(repository)
```

Comment:
Since `process_workspace` is the method that creates the `pant:workspace` jcr node, this should run regardless of the presence of any `attributeFile`.